File: data/multilang_comparison/program_julia/src/func24.c
#include "program_julia.h" 
array_t* func24(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array99 = func25(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array99->id);
      free(params0.data);
      array99->refC--;
      if(array99->refC == 0) {
         free(array99->data);
         free(array99);
         DEBUG_FREE(array99->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array109 = func26(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array109->id);
      free(params0.data);
      array109->refC--;
      if(array109->refC == 0) {
         free(array109->data);
         free(array109);
         DEBUG_FREE(array109->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array114 = func27(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array114->id);
   free(params0.data);
   return array114;
}

File: data/multilang_comparison/program_julia/src/func24.jl
include("program_julia_head.jl")
function func24(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      params0 = JArray_param{Int}()
      params0.size = 0
      array68 = func25(params0, get_path(), loopsFactor);
      array68.refC -= 1
      if array68.refC == 0
          empty!(array68.data)
          array68 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array77 = func26(params0, get_path(), loopsFactor);
      array77.refC -= 1
      if array77.refC == 0
          empty!(array77.data)
          array77 = nothing
      end
   end
   params0 = JArray_param{Int}()
   params0.size = 0
   array82 = func27(params0, get_path(), loopsFactor);
   return array82;
end

